### PR TITLE
Fix swallowed I/O exception in LcovPrinter

### PR DESCRIPTION
**Pull request by sputt**
This fixes issue #21982

### CODE REVIEW

File: tools/test/CoverageOutputGenerator/java/com/google/devtools/coverageoutputgenerator/LcovPrinter.java [hunk 1]
```diff
@@ -38,30 +38,18 @@ private LcovPrinter(BufferedWriter bufferedWriter) {
...
-      LcovPrinter lcovPrinter = new LcovPrinter(bufferedWriter);
-      lcovPrinter.print(coverage);
-      bufferedWriter.close();
-    } catch (IOException exception) {
-      logger.log(Level.SEVERE, "Could not write to output file.");
-      return false;
-    }
-    return true;
+  static void print(OutputStream outputStream, Coverage coverage) throws IOException {
+    Writer fileWriter = new OutputStreamWriter(outputStream, UTF_8);
```
Comment (c-mita) [inline tools/test/CoverageOutputGenerator/java/com/google/devtools/coverageoutputgenerator/LcovPrinter.java:42]: Should still keep the try-with-resource (or try-finally) to close the writer.

**Review by c-mita — CHANGES_REQUESTED** [review]

File: tools/test/CoverageOutputGenerator/java/com/google/devtools/coverageoutputgenerator/LcovPrinter.java (hunk 1 above)
Comment (sputt) [reply tools/test/CoverageOutputGenerator/java/com/google/devtools/coverageoutputgenerator/LcovPrinter.java]: Thanks, my java is rusty.  Added buffered writer in as well

**Review by c-mita — APPROVED** [review]
LGTM

**Comment by sputt** [comment]
Squashed to correct email - no diff.

## Functional Test

### Coverage roll-up
```
bazel coverage //tools/test/CoverageOutputGenerator/... --combined_report=lcov --override_repository=bazel_tools~remote_coverage_tools_extension~remote_coverage_tools=/Users/sputt/remote_coverage_tools
```
To sanity check this is the correct override:
```
ERROR: /private/var/tmp/_bazel_sputt/b1abbf127d696f148ad833cd4ef695e8/external/bazel_tools/tools/test/BUILD:36:6: @@bazel_tools//tools/test:coverage_report_generator depends on @@bazel_tools~remote_coverage_tools_extension~remote_coverage_tools//:coverage_report_generator in repository @@bazel_tools~remote_coverage_tools_extension~remote_coverage_tools which failed to fetch. no such package '@@bazel_tools~remote_coverage_tools_extension~remote_coverage_tools//': The repository's path is "bazel_tools~remote_coverage_tools_extension~remote_coverage_tools" (absolute: "/Users/sputt/remote_coverage_tools/JUNK") but it does not exist or is not a directory.
```

Inspect the output report:
```
ls -l bazel-out/_coverage/_coverage_report.dat 
-r-xr-xr-x  1 sputt  wheel  52981 Apr 13 09:23 bazel-out/_coverage/_coverage_report.dat
```

All files are present:
```
cat bazel-out/_coverage/_coverage_report.dat | grep SF:
SF:tools/test/CoverageOutputGenerator/java/com/google/devtools/coverageoutputgenerator/BranchCoverage.java
SF:tools/test/CoverageOutputGenerator/java/com/google/devtools/coverageoutputgenerator/Constants.java
SF:tools/test/CoverageOutputGenerator/java/com/google/devtools/coverageoutputgenerator/Coverage.java
SF:tools/test/CoverageOutputGenerator/java/com/google/devtools/coverageoutputgenerator/GcovJsonParser.java
SF:tools/test/CoverageOutputGenerator/java/com/google/devtools/coverageoutputgenerator/GcovParser.java
SF:tools/test/CoverageOutputGenerator/java/com/google/devtools/coverageoutputgenerator/LcovMergerFlags.java
SF:tools/test/CoverageOutputGenerator/java/com/google/devtools/coverageoutputgenerator/LcovParser.java
SF:tools/test/CoverageOutputGenerator/java/com/google/devtools/coverageoutputgenerator/LcovPrinter.java
SF:tools/test/CoverageOutputGenerator/java/com/google/devtools/coverageoutputgenerator/LineCoverage.java
SF:tools/test/CoverageOutputGenerator/java/com/google/devtools/coverageoutputgenerator/Main.java
SF:tools/test/CoverageOutputGenerator/java/com/google/devtools/coverageoutputgenerator/Parser.java
SF:tools/test/CoverageOutputGenerator/java/com/google/devtools/coverageoutputgenerator/SourceFileCoverage.java
```

**Comment by fmeum** [comment]
@bazel-io fork 7.2.0

**Pull request closed without merging**